File: src/api/ShowNTell.API/Models/Responses/LikeResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ShowNTell.API.Models.Responses
{
    /// <summary>
    /// A model for a like response from the API.
    /// </summary>
    public class LikeResponse
    {
        public int ImagePostId { get; set; }
        public string UserEmail { get; set; }
        public DateTime DateCreated { get; set; }
    }
}

File: src/show-n-tell-vue/src/store/plugins/signalr/responses/like-response.js
function LikeResponse(imagePostId, userEmail, username, dateCreated) {
    this.imagePostId = imagePostId
    this.userEmail = userEmail
    this.username = username
    this.dateCreated = new Date(dateCreated)
}

LikeResponse.fromJSON = function(like) {
    return new LikeResponse(
        like.imagePostId,
        like.userEmail,
        like.username,
        like.dateCreated
    )
}

export default LikeResponse